inference/core/workflows/core_steps/models/foundation/google_vision_ocr/v1.py

```diff
@@ -0,0 +1,181 @@
+from typing import List, Literal, Optional, Type, Union
+from uuid import uuid4
+
+import numpy as np
+import requests
+import supervision as sv
+from pydantic import ConfigDict, Field
+from supervision.config import CLASS_NAME_DATA_FIELD
+
+from inference.core.workflows.execution_engine.constants import (
+    DETECTION_ID_KEY,
+    IMAGE_DIMENSIONS_KEY,
+    PREDICTION_TYPE_KEY,
+)
+from inference.core.workflows.execution_engine.entities.base import (
+    OutputDefinition,
+    WorkflowImageData,
+)
+from inference.core.workflows.execution_engine.entities.types import (
+    OBJECT_DETECTION_PREDICTION_KIND,
+    STRING_KIND,
+    StepOutputImageSelector,
+    WorkflowImageSelector,
+    WorkflowParameterSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+LONG_DESCRIPTION = """
+Detect text in images using Google Vision OCR.
+
+Supported types of text detection:
+
+- `text_detection`: optimized for areas of text within a larger image.
+- `ocr_text_detection`: optimized for dense text documents.
+
+You need to provide your Google Vision API key to use this block.
+"""
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "Google Vision OCR",
+            "version": "v1",
+            "short_description": "Detect text in images using Google Vision API",
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "model",
+        },
+        protected_namespaces=(),
+    )
+    type: Literal["roboflow_core/google_vision_ocr@v1"]
+    image: Union[WorkflowImageSelector, StepOutputImageSelector]
```

Comment:
Could you please add pydantic fields descriptions, for image:

```
Field(
    description="The image to run OCR",
    examples=["$inputs.image", "$steps.cropping.crops"],
)
```
That would make UI to render descriptions and our docs to have nice examples - we generate this kind of pages for each block automatically: https://inference.roboflow.com/workflows/blocks/detections_consensus/#properties

Reply:
Thanks for the input! I'll make the necessary fixes. I was hoping for some guidance on the project's standards 😄.

The project was easier to contribute to than expected. The links in the issue and the docs on how to contribute / workflow docs were very helpful. 👏

Reply:
cool. Let's have that PR closed soon 😄